src/app/user.ts

```diff
@@ -1,12 +1,18 @@
+import {AppContext} from 'mattermost-redux/types/apps';
+import GeneralConstants from 'mattermost-redux/constants/general';
+
 import {newMMClient} from '../clients';
-import {oauthStore} from '../store';
+import {newTokenStore} from '../store';
 
-export function isUserConnected(userID: string): boolean {
-    return Boolean(oauthStore.getToken(userID));
+export async function isUserConnected(context: AppContext): Promise<boolean> {
+    const userID = context.acting_user_id;
+    const tokenStore = newTokenStore(context);
+    const token = await tokenStore.getToken(userID);
+    return Boolean(token);
 }
 
-export async function isUserSysadmin(userID: string): Promise<boolean> {
-    const client = newMMClient().asAdmin();
-    const user = await client.getUser(userID);
-    return user.roles.includes('system_admin');
+export async function isUserSysadmin(context: AppContext): Promise<boolean> {
```

Comment:
This is easier to check if you just expand the user on the calls. The Go code is even simpler, because you can call `IsSystemAdmin` over the user object. Nevertheless, the function is not specially complicated (not sure if there is any util in redux for that)

```
func (u *User) IsSystemAdmin() bool {
	return IsInRole(u.Roles, SYSTEM_ADMIN_ROLE_ID)
}

func IsInRole(userRoles string, inRole string) bool {
	roles := strings.Split(userRoles, " ")

	for _, r := range roles {
		if r == inRole {
			return true
		}
	}

	return false
}
```

Reply:
There is a redux util to check for sysadmin.  I don't have the ability to get expanded acting_user context when /bindings is called. I'm going to create a ticket for this issue and resolved sometime after that is working. I want to move a bit forward with other issues and get this merged.  Thanks for bringing this up

Reply:
ticket created in Jira for non-expanded manifes bindings call https://mattermost.atlassian.net/browse/MM-33447

Reply:
And here for Zendesk implementation request. https://mattermost.atlassian.net/browse/MM-33448